src/libraries/System.Text.Json/tests/Serialization/WriteValueTests.cs

```diff
@@ -68,5 +68,37 @@ public static void SerializeToWriterRoundTripEscaping()
 
             Assert.Equal(normalizedString, json);
         }
+
+        public class CustomClassToExceedMaxBufferSize
+        {
+            public CustomClassToExceedMaxBufferSize Internal1 { get; set; }
+            public CustomClassToExceedMaxBufferSize Internal2 { get; set; }
+
+            static string name = new string('a', 53687091);
+            public string GetName1 { get { return name; } }
+            public string GetName2 { get { return name; } }
```

Comment:
nit: Use expression bodied members for all these properties

```suggestion
            public string GetName2 => s_name;
```